spec/requests/alarm_spec.rb

```diff
@@ -0,0 +1,137 @@
+require 'rails_helper'
+
+RSpec.describe "Alarms", type: :request do
+  let(:user) do 
+    create(:user)
+  end
+  describe "GET #index" do
+    it "exists and responds" do
+      login_as(user)
+      get alarms_path
+      expect(response.status).to eq(200)
+    end
+  end
+
+  describe 'GET #new' do
+    it "assigns a new Alarm to @alarm" do
+       get new_alarm_path
+       expect(assigns(:alarm)).to be_a_new(Alarm)
+    end
+
+    it "renders the :new template" do
+      get new_alarm_path
+      expect(response).to render_template :new
+    end
+  end
+  
+  describe 'GET #edit' do
+    it "assigns a requested to @timezone" do
+      login_as(user)
+      alarm = create(:alarm, user: user)
+      get edit_alarm_path id: alarm
+      expect(assigns(:alarm)).to eq alarm
+      expect(assigns(:user)).to eq user
+    end
+
+    it "renders the :edit template" do
+      login_as(user)
+      alarm = create(:alarm)
+      get edit_alarm_path id: alarm
+      expect(response).to render_template :edit
+    end 
```

Comment:
Maybe also with the login_as

Reply:
Ok, ok I will try with that, my first thought was that it was necessary in the beginning of eac test case.